src/main/java/uk/gov/hmcts/ccd/domain/service/search/CreatorSearchOperation.java

```diff
@@ -38,8 +37,6 @@ public CreatorSearchOperation(@Qualifier("authorised") SearchOperation searchOpe
             return Lists.newArrayList();
         }
 
-        return results.stream()
-            .filter(caseAccessService::canUserAccess)
-            .collect(Collectors.toList());
+        return caseAccessService.filterByUserAccess(results);
```

Comment:
this is a very good improvement. However this solution still retrieves from the db potentially much more cases than needed. Because filtering based on user access happens in a second stage. I was wondering whether we could improve things even further by modifying the criteria passed to execute the very first query: 
```        
final List<CaseDetails> results = searchOperation.execute(metaData, criteria);
```
For users with restricted access to cases, we could as first thing check the list of case the user has access to, e.g.
```
 final List<Long> grantedCases = caseUserRepository.findCasesUserIdHasAccessTo(userRepository.getUserId());
```
Then enhance the criteria mechanism so that we can restrict a search on a range of cases only, in this example the `grantedCases`. 
And then we execute:
```
searchOperation.execute(metaData, criteriaWithRestrictedCases);
```